pkg/genericcontrolplane/clientutils/multiclusterconfig.go

```diff
@@ -88,11 +88,41 @@ func defaultRequestInfoResolver() genericapirequest.RequestInfoResolver {
 
 func (mcrt *multiClusterClientConfigRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	req = utilnet.CloneRequest(req)
+
+	requestClusterName := ""
+	if path := req.URL.Path; strings.HasPrefix(path, "/clusters/") {
+		path = strings.TrimPrefix(path, "/clusters/")
+		if i := strings.Index(path, "/"); i != -1 {
+			requestClusterName, path = path[:i], path[i:]
+			req.URL.Path = path
+			for i := 0; i < 2 && len(req.URL.RawPath) > 1; i++ {
+				if slash := strings.Index(req.URL.RawPath[1:], "/"); slash != -1 {
+					req.URL.RawPath = req.URL.RawPath[slash:]
+				}
+			}
+		}
+	} else {
+		requestClusterName = req.Header.Get("X-Kubernetes-Cluster")
+	}
```

Comment:
please add go doc oneliner on-top. This is non-trivial code. We shouldn't force every reader to read 15 lines of string manipulation.

Reply:
@ncdc is all of this going away with scopes? Then I don't care about ^ so much.

Reply:
FYI it's mostly a copy of what's in kcp's server.go WithClusterScope handler